src/test/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImplTest.java

```diff
@@ -486,21 +490,22 @@ private ObjectNode buildVersionRequest() {
     }
 
     private List<Block> buildBlocks() {
-        BlockHeader block1Header = blockHeaderBuilder.setNumber(1).build();
-        BlockHeader block2Header = blockHeaderBuilder.setNumber(2).build();
-        BlockHeader block3Header = blockHeaderBuilder.setNumber(3).build();
-
-        List<BlockHeader> block1Brothers = Arrays.asList(
-            blockHeaderBuilder.setNumber(101).build(),
-            blockHeaderBuilder.setNumber(102).build()
+        // 3 Blocks, 1 Brother Each For Block 2 And Block 1
+        List<BlockHeader> block1Uncles = Arrays.asList(
+            blockHeaderBuilder.setNumber(101).setParentHashFromKeccak256(TestUtils.createHash(0)).build(),
+            blockHeaderBuilder.setNumber(102).setParentHashFromKeccak256(TestUtils.createHash(0)).build()
+        );
+        List<BlockHeader> block2Uncles = Arrays.asList(
+            blockHeaderBuilder.setNumber(201).setParentHashFromKeccak256(TestUtils.createHash(1)).build(),
+            blockHeaderBuilder.setNumber(202).setParentHashFromKeccak256(TestUtils.createHash(0)).build()
         );
-        List<BlockHeader> block2Brothers = Collections.emptyList();
-        List<BlockHeader> block3Brothers = Collections.singletonList(blockHeaderBuilder.setNumber(301).build());
+        List<BlockHeader> block3Uncles = Collections.singletonList(
+            blockHeaderBuilder.setNumber(301).setParentHashFromKeccak256(TestUtils.createHash(2)).build());
 
         return Arrays.asList(
-            new Block(block1Header, Collections.emptyList(), block1Brothers, true, true),
-            new Block(block2Header, Collections.emptyList(), block2Brothers, true, true),
-            new Block(block3Header, Collections.emptyList(), block3Brothers, true, true)
+            TestUtils.mockBlockWithUncles(1, TestUtils.createHash(1), TestUtils.createHash(0), block1Uncles),
+            TestUtils.mockBlockWithUncles(2, TestUtils.createHash(2), TestUtils.createHash(1), block2Uncles),
+            TestUtils.mockBlockWithUncles(3, TestUtils.createHash(3), TestUtils.createHash(2), block3Uncles)
```

Comment:
Why go back to mocks? It's preferable to use the actual Block object, creates a more realistic environment. Block class constructor allows sending a list of uncles

Reply:
Updated